### PR TITLE
chore: Initial build, CI, and docs

### PR DESCRIPTION
**Pull request by patriknw**
* mostly copied from akka-persistence-r2dbc

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+        "https://github.com/akka/akka-persistence-dynamodb.git")),
+    startYear := Some(2024),
+    developers += Developer(
+      "contributors",
+      "Contributors",
+      "https://gitter.im/akka/dev",
+      url("https://github.com/akka/akka-persistence-dynamodb/graphs/contributors")),
+    releaseNotesURL := (
+      if (isSnapshot.value) None
+      else Some(url(s"https://github.com/akka/akka-persistence-dynamodb/releases/tag/v${version.value}"))
```
Comment (patriknw) [inline build.sbt:26]: I mostly searched and replaced r2dbc with dynamodb, so these links point to the wrong place. We'll decide later if it's going to be akka/akka-persistence-dynamodb 2.0 or a new repository, or kept private.

File: build.sbt [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+  Seq(
+    crossScalaVersions := Dependencies.ScalaVersions,
+    scalaVersion := Dependencies.Scala213,
+    crossVersion := CrossVersion.binary,
+    scalafmtOnCompile := true,
+    // Setting javac options in common allows IntelliJ IDEA to import them automatically
+    Compile / javacOptions ++= Seq("-encoding", "UTF-8", "--release", "11"),
+    Compile / scalacOptions ++= Seq("-release", "11"),
+    headerLicense := Some(
+      HeaderLicense.Custom("""Copyright (C) 2022 - 2023 Lightbend Inc. <https://www.lightbend.com>""")),
```
Comment (pvlugter) [inline build.sbt]: Not important but should we reset the copyright?

File: docs/src/main/paradox/getting-started.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  group=com.lightbend.akka
+  artifact=akka-persistence-dynamodb_$scala.binary.version$
+  version=$project.version$
+}
+
+This plugin depends on Akka $akka.version$ or later, and note that it is important that all `akka-*` 
+dependencies are in the same version, so it is recommended to depend on them explicitly to avoid problems 
+with transient dependencies causing an unlucky mix of versions.
+
+The plugin is published for Scala 2.13 and 2.12.
```
Comment (pvlugter) [inline docs/src/main/paradox/getting-started.md]: Probably also in the r2dbc docs too. Should be updated to scala 3 in some way.

```suggestion
The plugin is published for Scala 2.13 and 3.3.
```

File: project/Dependencies.scala [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+import sbt._
+
+object Dependencies {
+  val Scala213 = "2.13.13"
+  val Scala3 = "3.3.3"
+  val Scala2Versions = Seq(Scala213)
+  val ScalaVersions = Dependencies.Scala2Versions :+ Dependencies.Scala3
+  val AkkaVersion = System.getProperty("override.akka.version", "2.9.3")
+  val AkkaVersionInDocs = AkkaVersion.take(3)
```
Comment (pvlugter) [inline project/Dependencies.scala]: Just noticing while glancing through this: fragile for when 2.10 is released. Will probably be noticed here when updating though, but could use sbt version parsing.

**Review by pvlugter — APPROVED** [review]
LGTM. Useful to start with all the initial structure.

Build failing on sbt formatting.

File: docs/src/main/paradox/getting-started.md (hunk 3 above)
Comment (patriknw) [reply docs/src/main/paradox/getting-started.md]: fixed

File: project/Dependencies.scala (hunk 4 above)
Comment (patriknw) [reply project/Dependencies.scala]: fixed in 02a8b9ab52f8557af814608125b8ea817969e94e, also in r2dbc

**Pull request merged**